components/eventing-controller/api/v1alpha2/condition_unit_test.go

```diff
@@ -0,0 +1,450 @@
+// todo unit
+package v1alpha2_test
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha2"
+)
+
+func Test_InitializeSubscriptionConditions(t *testing.T) {
+	var tests = []struct {
+		name            string
+		givenConditions []v1alpha2.Condition
+	}{
+		{
+			name:            "Conditions empty",
+			givenConditions: v1alpha2.MakeSubscriptionConditions(),
+		},
+		{
+			name: "Conditions partially initialized",
+			givenConditions: func() []v1alpha2.Condition {
```

Comment:
let's remove the callback here:
```suggestion
			givenConditions: []v1alpha2.Condition{
				{
					Type:               v1alpha2.ConditionSubscribed,
					LastTransitionTime: metav1.Now(),
					Status:             corev1.ConditionUnknown,
				},
			},
```